arch/arm/core/aarch32/thread.c

```diff
@@ -29,112 +36,67 @@
  * of the ESF.
  */
 void arch_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
-		     size_t stackSize, k_thread_entry_t pEntry,
-		     void *parameter1, void *parameter2, void *parameter3,
-		     int priority, unsigned int options)
+		     char *stack_ptr, k_thread_entry_t entry,
+		     void *p1, void *p2, void *p3)
 {
-	char *pStackMem = Z_THREAD_STACK_BUFFER(stack);
-	char *stackEnd;
-	/* Offset between the top of stack and the high end of stack area. */
-	uint32_t top_of_stack_offset = 0U;
-
-#if defined(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT) \
-	&& defined(CONFIG_USERSPACE)
-	/* This is required to work-around the case where the thread
-	 * is created without using K_THREAD_STACK_SIZEOF() macro in
-	 * k_thread_create(). If K_THREAD_STACK_SIZEOF() is used, the
-	 * Guard size has already been take out of stackSize.
-	 */
-	stackSize -= MPU_GUARD_ALIGN_AND_SIZE;
-#endif
-
-#if defined(CONFIG_USERSPACE)
-	/* Truncate the stack size to align with the MPU region granularity.
-	 * This is done proactively to account for the case when the thread
-	 * switches to user mode (thus, its stack area will need to be MPU-
-	 * programmed to be assigned unprivileged RW access permission).
-	 */
-	stackSize &= ~(CONFIG_ARM_MPU_REGION_MIN_ALIGN_AND_SIZE - 1);
-
-#ifdef CONFIG_THREAD_USERSPACE_LOCAL_DATA
-	/* Reserve space on top of stack for local data. */
-	uint32_t p_local_data = Z_STACK_PTR_ALIGN(pStackMem + stackSize
-		- sizeof(*thread->userspace_local_data));
-
-	thread->userspace_local_data =
-		(struct _thread_userspace_local_data *)(p_local_data);
-
-	/* Top of actual stack must be moved below the user local data. */
-	top_of_stack_offset = (uint32_t)
-		(pStackMem + stackSize - ((char *)p_local_data));
-
-#endif /* CONFIG_THREAD_USERSPACE_LOCAL_DATA */
-#endif /* CONFIG_USERSPACE */
-
-#if defined(CONFIG_FPU) && defined(CONFIG_FPU_SHARING) \
-	&& defined(CONFIG_MPU_STACK_GUARD)
-	/* For a thread which intends to use the FP services, it is required to
-	 * allocate a wider MPU guard region, to always successfully detect an
-	 * overflow of the stack.
-	 *
-	 * Note that the wider MPU regions requires re-adjusting the stack_info
-	 * .start and .size.
-	 *
-	 */
-	if ((options & K_FP_REGS) != 0) {
-		pStackMem += MPU_GUARD_ALIGN_AND_SIZE_FLOAT
-			- MPU_GUARD_ALIGN_AND_SIZE;
-		stackSize -= MPU_GUARD_ALIGN_AND_SIZE_FLOAT
-			- MPU_GUARD_ALIGN_AND_SIZE;
+	struct __basic_sf *iframe;
+
+#ifdef CONFIG_MPU_STACK_GUARD
+#if defined(CONFIG_USERSPACE) && \
+		defined(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT)
+	if (z_stack_is_user_capable(stack)) {
+		/* Guard area is carved-out of the buffer, instead of reserved,
```

Comment:
@andrewboie is this now needed for the other ARM variants, too?
Just double-checking..

Reply:
Addressed now.
Missed this in testing as we lack an ARMv8 target with SPLIM disabled that runs tests/kernel/threads/thread_stack